[CI] PackageTests test32JavaHomeOverride failing

**Build Scans:**
- [elasticsearch-periodic-packaging #3525 / centos-7_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/a5szb2uanzldy)
- [elasticsearch-periodic-packaging #3525 / rhel-7_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/ekvzaqrutnao4)
- [elasticsearch-periodic-packaging #3525 / sles-12_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/usjxpddeda2mc)
- [elasticsearch-periodic-packaging #3522 / centos-7_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/eibnn6bdcqywg)
- [elasticsearch-periodic-packaging #3522 / rhel-7_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/xpdv5pbjdbqwu)
- [elasticsearch-periodic-packaging #3522 / sles-12_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/i2amdzo2psjaq)
- [elasticsearch-periodic-packaging #3519 / rhel-7_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/vcyxvdhtgrphe)
- [elasticsearch-periodic-packaging #3519 / centos-7_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/sdk3qdnmdl4lu)
- [elasticsearch-periodic-packaging #3519 / sles-12_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/qrm2dnfmw3g4e)
- [elasticsearch-periodic-packaging #3516 / centos-7_packaging-tests-unix](https://gradle-enterprise.elastic.co/s/nnxeq4gobgr4q)

**Reproduction Line:**
```
null
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test32JavaHomeOverride'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.packaging.test.PackageTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Startup command should succeed. Stderr: [exitCode = [1] stdout = [] stderr = [Job for elasticsearch.service failed because the control process exited with error code. See "systemctl status elasticsearch.service" and "journalctl -xe" for details.]]
Expected: <0>
     but: was <1>
```

**Issue Reasons:**
- [main] 2 consecutive failures in test test32JavaHomeOverride
- [main] 8 consecutive failures in step centos-7_packaging-tests-unix
- [main] 9 consecutive failures in step rhel-7_packaging-tests-unix
- [main] 9 consecutive failures in step sles-12_packaging-tests-unix
- [main] 26 failures in test test32JavaHomeOverride (9.2% fail rate in 282 executions)
- [main] 8 failures in step centos-7_packaging-tests-unix (100.0% fail rate in 8 executions)
- [main] 9 failures in step rhel-7_packaging-tests-unix (100.0% fail rate in 9 executions)
- [main] 9 failures in step sles-12_packaging-tests-unix (100.0% fail rate in 9 executions)
- [main] 9 failures in pipeline elasticsearch-periodic-packaging (100.0% fail rate in 9 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/libs/native/jna/src/main/java/org/elasticsearch/nativeaccess/jna/JnaNativeLibraryProvider.java
+++ b/libs/native/jna/src/main/java/org/elasticsearch/nativeaccess/jna/JnaNativeLibraryProvider.java
@@ -8,6 +8,7 @@
 
 package org.elasticsearch.nativeaccess.jna;
 
+import org.elasticsearch.core.SuppressForbidden;
 import org.elasticsearch.nativeaccess.lib.JavaLibrary;
 import org.elasticsearch.nativeaccess.lib.Kernel32Library;
 import org.elasticsearch.nativeaccess.lib.LinuxCLibrary;
@@ -25,7 +26,7 @@ import java.util.function.Supplier;
 public class JnaNativeLibraryProvider extends NativeLibraryProvider {
 
     static {
-        System.setProperty("jna.library.path", LoaderHelper.platformLibDir.toString());
+        setJnaLibraryPath();
     }
 
     public JnaNativeLibraryProvider() {
@@ -48,6 +49,11 @@ public class JnaNativeLibraryProvider extends NativeLibraryProvider {
                 notImplemented()
             )
         );
+    }
+
+    @SuppressForbidden(reason = "jna library path must be set for load library to work with our own libs")
+    private static void setJnaLibraryPath() {
+        System.setProperty("jna.library.path", LoaderHelper.platformLibDir.toString());
     }
 
     private static Supplier<NativeLibrary> notImplemented() {

--- a/libs/native/jna/src/main/java/org/elasticsearch/nativeaccess/jna/JnaNativeLibraryProvider.java
+++ b/libs/native/jna/src/main/java/org/elasticsearch/nativeaccess/jna/JnaNativeLibraryProvider.java
@@ -11,6 +11,7 @@ package org.elasticsearch.nativeaccess.jna;
 import org.elasticsearch.nativeaccess.lib.JavaLibrary;
 import org.elasticsearch.nativeaccess.lib.Kernel32Library;
 import org.elasticsearch.nativeaccess.lib.LinuxCLibrary;
+import org.elasticsearch.nativeaccess.lib.LoaderHelper;
 import org.elasticsearch.nativeaccess.lib.MacCLibrary;
 import org.elasticsearch.nativeaccess.lib.NativeLibrary;
 import org.elasticsearch.nativeaccess.lib.NativeLibraryProvider;
@@ -22,6 +23,10 @@ import java.util.Map;
 import java.util.function.Supplier;
 
 public class JnaNativeLibraryProvider extends NativeLibraryProvider {
+
+    static {
+        System.setProperty("jna.library.path", LoaderHelper.platformLibDir.toString());
+    }
 
     public JnaNativeLibraryProvider() {
         super(

--- a/libs/native/src/main/java/org/elasticsearch/nativeaccess/lib/LoaderHelper.java
+++ b/libs/native/src/main/java/org/elasticsearch/nativeaccess/lib/LoaderHelper.java
@@ -16,7 +16,7 @@ import java.nio.file.Paths;
  * A utility for loading libraries from Elasticsearch's platform specific lib dir.
  */
 public class LoaderHelper {
-    private static final Path platformLibDir = findPlatformLibDir();
+    public static final Path platformLibDir = findPlatformLibDir();
 
     private static Path findPlatformLibDir() {
         // tests don't have an ES install, so the platform dir must be passed in explicitly

